### PR TITLE
Adding publish to context menu

### PR DESCRIPTION
**Pull request by vshan**

### CODE REVIEW

File: src/commands/publish-application.ts [hunk 1]
```diff
@@ -88,8 +88,12 @@ async function installApplication(terminal:vscode.Terminal) {
 async function readCloudProfile() {
     var fs = require('fs');
     const cloudProfile: vscode.Uri[] = await vscode.workspace.findFiles('**/Cloud.json');
-    const pathToCloudProfile = cloudProfile[0].fsPath.replace('/c:', '');
+    if (cloudProfile.length < 1) {
+        vscode.window.showErrorMessage("Could not find configuration file Cloud.json. Please ensure you have run the build command.");
```
Comment (anantshankar17) [inline src/commands/publish-application.ts]: Please ensure that the application package is built using the build command before executing publish.

File: src/commands/publish-application.ts [hunk 2]
```diff
@@ -88,8 +88,12 @@ async function installApplication(terminal:vscode.Terminal) {
 async function readCloudProfile() {
     var fs = require('fs');
     const cloudProfile: vscode.Uri[] = await vscode.workspace.findFiles('**/Cloud.json');
-    const pathToCloudProfile = cloudProfile[0].fsPath.replace('/c:', '');
+    if (cloudProfile.length < 1) {
```
Comment (anantshankar17) [inline src/commands/publish-application.ts:91]: is this check present in deploy cmd ?

**Review by anantshankar17 — APPROVED** [review]

File: src/commands/publish-application.ts (hunk 2 above)
Comment (vshan) [reply src/commands/publish-application.ts:91]: In publish we check for the presence of Cloud.json, and In deploy we check for the presence of install.ps1 and return error messages accordingly. However, install.ps1 is created during application creation but cloud.json is created during application build time. 
So when deploy is run before build but after create, the commands in install.ps1 fail with 'could not find application packages' etc.

File: src/commands/publish-application.ts (hunk 1 above)
Comment (vshan) [reply src/commands/publish-application.ts]: Thanks Anant, have made the change

**Pull request merged**